src/LDAP/Search/Filter/Binary.hs

```diff
@@ -121,20 +120,24 @@ leadKeychar = alpha
 keychar :: Get Char
 keychar = asum [ alpha, digit, hyphen ]
 
-utf1Subset :: Get Char
-utf1Subset = Char8.satisfy $ not . (`elem` ['\NUL', '(', ')', '*', '\ESC'])  -- fixme: this parses 129-255 too
-
+-- Idea: the escaped portion of the valueEncoding and the subset exceptions
+-- only apply to ASCII code points. We can thus process it first without any
+-- concern for multi-byte encoding, then process the UTF-8 byte string as a
+-- whole.
 valueEncoding :: Get Text
-valueEncoding = Text.pack <$> many (normal <|> escaped)
+valueEncoding = toText <$> many valueEncodingByte
     where
-        normal = utf1Subset
+        -- 8-bit chars to Text. TODO: handle decoding error
```

Comment:
> 8-bit chars to Text

Better to express this with a type sig.

> TODO: handle decoding error

Let's rather use [`Data.Text.Encoding.decodeUtf8'`](https://hackage.haskell.org/package/text-2.0.1/docs/Data-Text-Encoding.html#g:3)

Reply:
Agreed! I didn't go that way at first because it requires building a ByteString. I just found out about [Data.ByteString.Char8](https://hackage.haskell.org/package/bytestring-0.9.2.1/docs/Data-ByteString-Char8.html) which should fit well with the existing byte parsers `normal` and `escaped`.